engine/unity5/Assets/Scripts/DriverPractice/Interactor.cs

```diff
@@ -37,13 +39,11 @@ private void Awake()
         /// <param name="manifoldList">List of collision manifolds--this isn't used</param>
         public void BOnCollisionEnter(CollisionObject other, BCollisionCallbacksDefault.PersistentManifoldList manifoldList)
         {
-            string gamepiece;
             for (int i = 0; i < collisionKeyword.Length; i++)
             {
-                if (collisionKeyword[i] != null)
+                if (collisionKeyword[i] != "NOT CONFIGURED" && collisionKeyword[i] != "NOT DEFINED" && collisionKeyword[i] != null && GameObject.Find(collisionKeyword[i]) != null)
                 {
-                    gamepiece = collisionKeyword[i];
-                    if (other.UserObject.ToString().Contains(gamepiece))
+                    if (Auxiliary.FindObject(Auxiliary.FindObject("Field"), collisionKeyword[i]).GetComponent<BRigidBody>().mass == ((BRigidBody)other.UserObject).mass && Auxiliary.FindObject(Auxiliary.FindObject("Field"), collisionKeyword[i]).GetComponent<BRigidBody>().friction == ((BRigidBody)other.UserObject).friction && Auxiliary.FindObject(Auxiliary.FindObject("Field"), collisionKeyword[i]).GetComponent<BRigidBody>().rollingFriction == ((BRigidBody)other.UserObject).rollingFriction)
```

Comment:
This is a lot to process every time a collision occurs. Especially since the game element's GameObject and components are being searched for when comparing each individual property. I think a more efficient version of this method would be to find the required components once and then check the properties using that established reference. Also, what if two different property sets have all the same friction and mass values but contain different field elements? Wouldn't this `if` statement pass? I think maybe a different approach might be to somehow attach the property set ID to the field elements on field generation. Then you could compare these IDs when the collision occurs. That would have a lot less overhead. Every GameObject has a "tag" property so you could probably set that property to the property set ID and check the tag when the collision occurs.